cmd/kube-apiserver/app/options/options.go

```diff
@@ -93,6 +93,12 @@ type ServerRunOptions struct {
 	ServiceAccountTokenMaxExpiration time.Duration
 
 	ShowHiddenMetricsForVersion string
+
+	// AssumePrunedCustomResourceMetaDataInStorage set to true optimizes the read code-path for CRs
+	// assuming that ObjectMeta has been pruned when writing the objects. This can be
+	// assumed for every cluster installed with or after 1.11 when ObjectMeta pruning
+	// was added.
+	AssumePrunedCustomResourceMetaDataInStorage bool
```

Comment:
would it be terrible to default this to deriving the setting per-CRD from the creation timestamp of the CRD? 

for example, if a CRD's creation timestamp was well after the 1.10 EOL date (https://github.com/kubernetes/website/blob/main/content/en/releases/patch-releases.md#non-active-branch-history), like 2020-01-01